applications/ananke/src/ananke_vm_callback.erl

```diff
@@ -0,0 +1,270 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2015, 2600Hz
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%     SIPLABS, LLC (Ilya Ashchepkov)
+%%%-------------------------------------------------------------------
+-module(ananke_vm_callback).
+
+-export([init/0
+         ,handle_req/2
+         ,check/2
+        ]).
+
+-include("ananke.hrl").
+
+-record(args, {account_id
+               ,user_id
+               ,vm_box_id
+               ,callback_number
+               ,is_callback_disabled
+               ,vm_number
+               ,attempts
+               ,interval
+               ,call_timeout
+               ,realm
+              }).
+
+-spec init() -> 'ok'.
+init() ->
+    lager:debug("starting supervisor"),
+    {'ok', _} = ananke_sup:start_supervisor('ananke_callback_sup'),
+    'ok'.
+
+-spec check(ne_binary(), ne_binary()) -> any().
+check(AccountId, VMBoxId) ->
+    lager:info("checking vmbox ~p in ~p", [VMBoxId, AccountId]),
+    AccountDb = wh_util:format_account_id(AccountId, 'encoded'),
+    {'ok', VMBox} = couch_mgr:open_cache_doc(AccountDb, VMBoxId),
+    case [X || X <- wh_json:get_value(<<"messages">>, VMBox, [])
+               , wh_json:get_value(<<"folder">>, X) =:= <<"new">>]
+    of
+        [] ->
+            lager:info("no unreaded messages"),
+            'ok';
+        _  ->
+            lager:info("found unreaded messages"),
+            handle_req(wh_json:set_values([{<<"Account-ID">>, AccountId}
```

Comment:
I typically use `wh_json:from_list([...])` instead of `wh_json:set_values([...], wh_json:new())`; I think it reads better the intent - creating a new json object.